Framework/PCFileCreator.m

```diff
@@ -286,6 +286,29 @@ - (BOOL)createFiles:(NSDictionary *)fileList
   return YES;
 }
 
+- (NSString *) sanitizeUppercaseFileNameString:(NSString *) input
+{
+    NSMutableString *sanitizedString = [NSMutableString stringWithCapacity:[input length]];
+    
+    // Create a character set that includes letters, digits, and the underscore character
+    NSMutableCharacterSet *allowedCharacters = [NSMutableCharacterSet alphanumericCharacterSet];
+    [allowedCharacters addCharactersInString:@"_"];
+    
+    for (NSUInteger i = 0; i < [input length]; i++)
+      {
+        unichar character = [input characterAtIndex:i];
+        
+        if ([allowedCharacters characterIsMember:character])
+          {
+            [sanitizedString appendFormat:@"%C", character];
+          } else {
```

Comment:
This else should be on separate lines... like so...

  }
else
  {

Other than that looks good.

Reply:
done!